### DISCUSSION TITLE
Per-Keyboard XKB Configs

### DISCUSSION THREAD

**Discussion by arutonee1** [YaLTeR/niri#355] [Ideas]
Would it be possible to make per-keyboard xkb configs? I have a standard 65% QWERTY keyboard that I mapped to Dvorak using xkb and a custom 8-key macropad with my own layout flashed to it. Would it be possible to apply the xkb layer to only certain keyboards (my 65%) and not to others (my macropad), similar to "Per-device input configs" in Hyprland? Thanks!

**Comment by YaLTeR** [comment]
Per-device input config is planned, but not implemented yet unfortunately

**Reply by YaLTeR** [reply]
I guess per-device XKB settings will be slightly more complex, but doable I think
